app/models/solidus_stripe/gateway.rb

```diff
@@ -56,6 +94,41 @@ def capture(amount_in_cents, payment_intent_id, options = {})
       )
     end
 
+    # Authorizes and captures a certain amount on the provided payment source.
+    #
+    # See `#authorize` for how the confirmation step is performed.
+    #
+    # @todo add support for purchasing custom amounts
+    def purchase(amount_in_cents, _source, options = {})
+      check_given_amount_matches_payment_intent(amount_in_cents, options)
+
+      stripe_payment_intent_options = {
+        amount: to_stripe_amount(amount_in_cents, options[:originator].currency),
+        confirm: false,
+        capture_method: "automatic",
+      }
+
+      stripe_payment_intent = SolidusStripe::PaymentIntent.prepare_for_payment(
+        options[:originator],
+        **stripe_payment_intent_options,
+      )
+
+      confirm_stripe_payment_intent(stripe_payment_intent.stripe_intent_id)
```

Comment:
I think we're missing the capture part, right? :thinking:

Reply:
This is the `purchase` action; the capture should be part of the confirmation 🤔
With the automatic capture_mode set up, after the confirmation, we'll have an authorized+captured payment

Reply:
Oh, yeah, you're right. Sorry for the noise